packages/sshnoports/bundles/shell/install.sh

```diff
@@ -0,0 +1,296 @@
+  echo "debug           - install all debug binaries"
+  echo ""
+  echo "all             - install all binaries (base and debug)"
+  echo ""
+  echo "systemd <unit>  - install a systemd unit"
+  echo "                  available units: [sshnpd, sshrvd]"
+  echo ""
+  echo "headless <job>  - install a headless cron job"
+  echo "                  available jobs: [sshnpd, sshrvd]"
+  echo ""
+  echo "tmux <service>  - install a service in a tmux session"
+  echo "                  available services: [sshnpd, sshrvd]"
+}
+
+# INSTALL BINARIES #
+
+install_single_binary() {
+  if is_root; then
+    dest="$bin_dir"
+  else
+    dest="$user_bin_dir"
+  fi
+  mkdir -p "$dest"
+  cp "$script_dir/$1" "$dest/$1"
+  echo "Installed $1 to $dest"
+}
+
+install_base_binaries() {
+  install_single_binary "at_activate"
+  install_single_binary "sshnp"
+  install_single_binary "sshnpd"
+  install_single_binary "sshrv"
+  install_single_binary "sshrvd"
+}
+
+install_debug_binary() {
+  if is_root; then
+    dest="$bin_dir"
+  else
+    dest="$user_bin_dir"
+  fi
+  mkdir -p "$dest"
+  cp "$script_dir/debug/$1" "$bin_dir/debug_$1"
+  echo "Installed debug_$1 to $dest"
+}
+
+install_debug_binaries() {
+  install_debug_binary "sshrvd"
+}
+
+install_all_binaries() {
+  install_base_binaries
+  install_debug_binaries
+}
+
+# SYSTEMD #
+
+post_systemd_message() {
+  echo "Systemd unit installed, make sure to configure the unit by editing $dest"
+  echo "Learn more in $script_dir/systemd/README.md"
+  echo ""
+  echo "To enable the service on next boot:"
+  echo "  sudo systemctl enable $unit_name"
+  echo ""
+  echo "To start the service immediately:"
+  echo "  sudo systemctl start $unit_name"
+}
+
+install_systemd_unit() {
+  unit_name="$1"
+  no_mac
+  dest="$systemd_dir/$unit_name"
+  cp "$script_dir/systemd/$unit_name" "$dest"
+  post_systemd_message
+}
+
+install_systemd_sshnpd() {
+  root_only
+  install_single_binary "sshnpd"
+  install_single_binary "sshrv"
+  install_systemd_unit "sshnpd.service"
+}
+
+install_systemd_sshrvd() {
+  root_only
+  install_single_binary "sshrvd"
+  install_systemd_unit "sshrvd.service"
+}
+
+systemd() {
+  case "$1" in
+    --help) usage; exit 0;;
+    sshnpd) install_systemd_sshnpd;;
+    sshrvd) install_systemd_sshrvd;;
+    *)
+      echo "Unknown systemd unit: $1";
+      usage;
+      exit 1;
+  esac
+}
+
+# HEADLESS SERVICES #
+
+post_headless_message() {
+  echo "Headless job installed, make sure to configure the job by editing $dest"
+  echo "Learn more in $script_dir/headless/README.md"
+  echo ""
+  echo "The job will start upon next system boot"
+  echo "To start the job immediately:"
+  echo "  $command &"
+  echo ""
+  echo "Warning: logs stored at $log_file and $err_file have unlimited potential size"
+  echo "In a production environment, it's recommended that you install via systemd"
+  echo "or use a cron job to call logrotate on these log files if systemd is not available"
```

Comment:
logrotate isn't available on every system, so I'm not trying to automate it or have to call it... this option should be a last resort anyways.